### ISSUE TITLE
HTTP::WebSocket.new(URI | String) ignores the query part of a given URL

### ISSUE THREAD

**Issue by meew0** [crystal-lang/crystal#3234]
Example code that exhibits this behaviour (unfortunately not perfectly reproducible as I don't know of a public websocket testing server that uses query parameters - if you're testing this, replace the given one with one that does):

``` crystal
require "http/web_socket"

websocket = HTTP::WebSocket.new(URI.parse("wss://websocket.example.com/some_path?key=value"))

# The connection will now behave as if the `?key=value` wasn't there and, depending on the server, completely fail, behave wrongly or work perfectly fine.
websocket.run
```

I believe the relevant code is at https://github.com/crystal-lang/crystal/blob/d493f0e7e64e36c8c8eef520657a96906380e2b5/src/http/web_socket/protocol.cr#L269, where `new(String, String, Int32, Bool)` is called without appending the URL query string to the `path` part. A potential (untested) fix could be:

``` crystal
    if (host = uri.host) && (path = uri.path)
      tls = uri.scheme == "https" || uri.scheme == "wss"
      path += uri.query if uri.query
      return new(host, path, uri.port, tls)
    end
```

Although I believe the version information is irrelevant, I'm on Arch Linux running Crystal 0.18.7.

**Comment by jackturnbull** [comment]
Good/accurate report, just hit this myself in fact. I believe the fix itself can be slightly lighter:

```crystal
if (host = uri.host) && (path = uri.full_path)
  tls = uri.scheme == "https" || uri.scheme == "wss"
  return new(host, path, uri.port, tls, headers)
end
```

Where `(path = uri.path)` become `(path = uri.full_path)` on the first line.

**Comment by RX14** [comment]
Closed in #5095
